test/factory/factory-server.js

```diff
@@ -0,0 +1,28 @@
+'use strict'
+
+const Hapi = require('hapi')
+
+const port = Number(process.env.PORT) || 55155
+const options = {
+  connections: {
+    routes: {
+      cors: true
+    }
+  }
+}
+
+module.exports = (callback) => {
+  const http = new Hapi.Server(options)
```

Comment:
I like `hapi` because then I can just set CORS to true without having to write headers by hand. Also `hapi` is pretty modular, a lot of features (like template engine) are buy in modules, which makes it a 'smaller' framework